proto/connectrpc/conformance/v1/client_compat.proto

```diff
@@ -114,41 +136,53 @@ message ClientCompatRequest {
 
 // The outcome of one ClientCompatRequest.
 message ClientCompatResponse {
+  // The test name that this response applies to.
   string test_name = 1;
+  // These fields determine the outcome of the request. Note that an `error`
+  // should only be reported for unexpected internal errors. RPC errors, such
+  // as Connect error codes, should be reported as part of the `response`.
   oneof result {
     ClientResponseResult response = 2;
     ClientErrorResult error = 3;
   }
 }
 
 // The result of a ClientCompatRequest, which may or may not be successful.
+// The client will build this message and return it back to the test runner.
 message ClientResponseResult {
+  // All response headers read from the response.
   repeated Header response_headers = 1;
+  // Servers should echo back payloads that they received as part of the request.
+  // This field should contain all the payloads the server echoed back. Note that
+  // There will be zero-to-one for unary and client-stream methods and 
+  // zero-to-many for server- and bidi-stream methods.
   repeated ConformancePayload payloads = 2;
   // The error received from the actual RPC invocation. Note this is not representative
-  // of a runtime error and should always be the proto equivalent of a Connect error.
+  // of a runtime error and should always be the proto equivalent of a Connect
+  // or gRPC error.
   Error error = 3;
+  // All response headers read from the response.
   repeated Header response_trailers = 4;
-
   // The number of messages that were present in the request but that could not be
   // sent because an error occurred before finishing the upload.
   int32 num_unsent_requests = 5;
-
   // The following field is only set by the reference client. It communicates
   // the underlying HTTP status code of the server's response.
+  // If you are implementing a client-under-test, you may ignore this field.
   optional int32 http_status_code = 6;
-
   // This field is used only by the reference client, and it can be used
   // to provide additional feedback about problems observed in the server
   // response or in client processing of the response. If non-empty, the test
   // case is considered failed even if the result above matches all expectations.
+  // If you are implementing a client-under-test, you may ignore this field.
   repeated string feedback = 7;
 }
 
 // The client is not able to fulfill the ClientCompatRequest. This may be due
 // to a runtime error or an unexpected internal error such as the requested protocol
 // not being supported. This is completely independent of the actual RPC invocation.
 message ClientErrorResult {
+  // A message describing the error that occurred.
```

Comment:
Maybe mention that this string will be shown to users running conformance tests so should include any relevant details that may help them troubleshoot or remedy the error.